lib/capdrupal.rb

```diff
@@ -0,0 +1,310 @@
+        end
+      end
+    end
+
+    namespace :backup do
+      desc "Check required files and directories exist"
+      task :check do
+        on release_roles :all do
+          execute :mkdir, "-p", "#{shared_path}/#{fetch(:backup_path)}"
+        end
+      end
+
+      desc "Clean up old drupal site database backup"
+      task :cleanup do
+        on roles(:app) do
+          within release_path.join(fetch(:app_path)) do
+            backup_path = "#{shared_path}/#{fetch(:backup_path)}"
+            keep_backups = fetch(:keep_backups, 5)
+
+            # Fetch every file from the backup directory, oldest on top.
+            backups = capture(:ls, "-ltrx", backup_path).split
+            info "Keeping #{keep_backups} of #{backups.count} backup dump on #{backup_path}."
+
+            # If we found less file than the keep number finish the task here
+            next "" unless backups.count > keep_backups.to_i
+
+            # Calculate number of file to delete.
+            to_delete = backups.count - keep_backups.to_i
+
+            # Loop throught file to delete (oldest files on top).
+            backups[0, to_delete].map do |backup|
+              execute :rm, "#{backup_path}/#{backup}"
+            end
+          end
+        end
+      end
+    end
+  end
+
+  namespace :cache do
+    desc 'Clear all caches'
+    task :clear do
+      on roles(:app) do
+        within release_path.join(fetch(:app_path)) do
+          execute :drush, 'cr'
+        end
+      end
+    end
+  end
+
+  namespace :module do
+    desc 'Enable module(s)'
+    task :enable do
+      on roles(:app) do
+        within release_path.join(fetch(:app_path)) do
+          for mod in fetch(:enable_modules)
+            execute :drush, "en '#{mod}' -y"
+          end
+        end
+      end
+    end
+
+    desc 'Disable module(s)'
+    task :disable do
+      on roles(:app) do
+        within release_path.join(fetch(:app_path)) do
+          for mod in fetch(:disable_modules)
+            execute :drush, "pmu '#{mod}' -y"
+          end
+        end
+      end
+    end
+  end
+
+  namespace :maintenance do
+    desc "Set maintenance mode"
+    task :on do
+      on roles(:app) do
+        within release_path.join(fetch(:app_path)) do
+          execute :drush, "state:set system.maintenance_mode 1 -y"
+          execute :drush, 'cr'
+        end
+      end
+    end
+
+    desc "Remove maintenance mode"
+    task :off do
+      on roles(:app) do
+        within release_path.join(fetch(:app_path)) do
+          execute :drush, "state:set system.maintenance_mode 0 -y"
+          execute :drush, 'cr'
+        end
+      end
+    end
+  end
+
+  desc 'Update database with migrations scripts'
+  task :updatedb do
+    on roles(:app) do
+      within release_path.join(fetch(:app_path)) do
+        execute :drush, 'updatedb -y'
+      end
+    end
+  end
+
+  namespace :updatedb do
+    desc 'Update database with migrations scripts'
```

Comment:
```suggestion
    desc 'Update database with migrations scripts (could silently fail)'
```

@WengerK @yannlugrin We always use this one (`drupal:updatedb:silence`). Does it make sense to keep the other one (`drupal:updatedb`) ?

Reply:
yes, the first allow a fail we use it only once, and then we use the second - that will break ci on fail (what we want).

We have to do that because Drupal is kind of dumb with update. Sometimes you only need to updateb & cim & sometimes you should cim updatedb cim updatedb. 

Therefore, we chose to updatedb once (may fail cause we dont cim before) then cim and finaly updatedb final. The last undatedb should never fail oterwise the website is not functional and we should rollback. That's why we have both

Reply:
We use both, and if the updatedb fail after config import, it's important that deploy fails.